provider/dual/options.go

```diff
@@ -118,6 +124,37 @@ func WithKeystore(ks keystore.Keystore) Option {
 	}
 }
 
+func withDatastore(ds datastore.Batching, dhts ...uint8) Option {
+	return func(cfg *config) error {
+		if ds == nil {
+			return errors.New("provider config: datastore cannot be nil")
+		}
+		switch len(dhts) {
+		case 1:
+			cfg.datastores[dhts[0]] = ds
+		case 2:
+			for _, dht := range dhts {
+				cfg.datastores[dht] = namespace.Wrap(ds, datastore.NewKey(descriptors[dht]))
+			}
+		default:
+			return errors.New("provider config: invalid number of dhts specified")
+		}
+		return nil
+	}
+}
+
+func WithDatastore(ds datastore.Batching) Option {
+	return withDatastore(ds, lanID, wanID)
+}
+
+func WithDatastoreLAN(ds datastore.Batching) Option {
+	return withDatastore(ds, lanID)
+}
+
+func WithDatastoreWAN(ds datastore.Batching) Option {
+	return withDatastore(ds, lanID)
```

Comment:
@guillaumemichel should this be wanID instead?
```suggestion
func WithDatastoreWAN(ds datastore.Batching) Option {
	return withDatastore(ds, wanID)
```